packages/proxy/lib/http/response-middleware.ts

```diff
@@ -145,6 +145,17 @@ const stringifyFeaturePolicy = (policy: any): string => {
   return pairs.map((directive) => directive.join(' ')).join('; ')
 }
 
+const getOriginalRequestId = (requestId: string) => {
+  let originalRequestId = requestId
+  const match = /^(.*)-retry-([\d]+)$/.exec(requestId)
```

Comment:
Maybe hoist this RegExp up so it's not recreated on every call?

Reply:
Fixed here: [`c7f795f` (#27771)](https://github.com/cypress-io/cypress/pull/27771/commits/c7f795fd98eee55554ca055eb33f9ad5a56ba941)